R/shinyui.R

```diff
@@ -80,6 +60,40 @@ renderPage <- function(ui, showcase=0, testMode=FALSE) {
   enc2utf8(paste(collapse = "\n", html))
 }
 
+shinyDependencies <- tagFunction(function() {
+  cssFile <- shinyCssFile()
+
+  version <- utils::packageVersion("shiny")
+  list(
+    htmlDependency(
+      name = "shiny-css",
+      version = version,
+      src = cssFile$src,
+      stylesheet = cssFile$stylesheet
+    ),
+    htmlDependency(
+      name = "shiny-javascript",
+      version = version,
+      src = c(href = "shared"),
+      script = if (getOption("shiny.minified", TRUE)) "shiny.min.js" else "shiny.js"
+    )
+  )
+})
+
+shinyCssFile <- function(theme = getShinyOption("bootstrapTheme")) {
+  if (!is_bs_theme(theme)) {
+    return(list(src = c(href = "shared"), stylesheet = "shiny.css"))
+  }
+  scss_home <- system.file("www", "shared", "shiny_scss", package = "shiny")
+  scss_files <- file.path(scss_home, c("bootstrap.scss", "shiny.scss"))
+  outFile <- bootstrapSass(lapply(scss_files, sass::sass_file), theme = theme, "shiny")
```

Comment:
We can address this in a separate PR, but it feels a little funny that we have to manually construct the html dependency object for this.

`bs_dependencies` does that for the main bootstrap content, and it would be nice if we could do something similar for cases like this.

Reply:
Sounds good, let's do it in a follow up PR